src/components/workAllocation/WorkRow.tsx

```diff
@@ -84,12 +88,64 @@ export default function WorkRow({
     });
   };
 
+  /**
+   * Callback for when the user edits Number of blocks or slides in the table/>
+   */
+  const handleWorkNumValueChange = useCallback(
+    (workNumValue: string | number, workNumValueType: string) => {
+      let value = workNumValue === "" ? undefined : Number(workNumValue);
+      if (value && value > MAX_NUM_BLOCKANDSLIDES)
+        value = MAX_NUM_BLOCKANDSLIDES;
+      if (workNumValueType === "block") {
+        send({ type: "UPDATE_NUM_BLOCKS", numBlocks: value });
+      } else {
+        send({ type: "UPDATE_NUM_SLIDES", numSlides: value });
+      }
+    },
+    [send]
+  );
+
+  const renderWorkNumValueField = (
+    workNumber: string,
+    workNumValue: number | undefined,
+    workNumValueType: string
+  ) => {
+    return (
+      <input
+        data-testid={workNumber + "-" + workNumValueType}
+        className={"border-0 border-gray-100 "}
+        type="number"
+        min="0"
+        max={MAX_NUM_BLOCKANDSLIDES}
+        step="1"
+        onChange={(e) => {
+          handleWorkNumValueChange(e.currentTarget.value, workNumValueType);
+        }}
+        value={workNumValue ?? ""}
```

Comment:
I think React will give a warning about controlled/uncontrolled inputs about this. May need to use `defaultValue` instead.